src/NBitcoin/StandardScriptTemplate.cs

```diff
@@ -51,22 +51,22 @@ protected override bool FastCheckScriptPubKey(Script scriptPubKey, out bool need
             needMoreCheck = true;
             return true;
         }
-        protected override bool CheckScriptPubKeyCore(Script scriptPubKey, Op[] scriptPubKeyOps)
+        protected override bool CheckScriptPubKeyCore(Network network, Script scriptPubKey, Op[] scriptPubKeyOps)
```

Comment:
why network here?

Reply:
i don't mean only here, it is all over the place

Reply:
yes that because of serialization at the lower level of the code and the way the code is intermingled with each other.
notice this an override, there must have been a place where the consensus factory was required.
I agree this could be designed better but that would mean a much bigger change.

Reply:
That is what I'm asking about – how is this connected to serialisation? It cannot see any connection, although it is super easy to miss one because the code is too messy. So could you show me why is this needed?

Reply:
I am trying to find the reason why I had to change the ScriptPolicy to add network

Reply:
Ah the override method `CheckScriptSigCore` has two places where its calling `TransactionSignature.IsValid` this requires the network as signature verification is happening.

I guess the network could go away from `CheckScriptPubKeyCore` (trying to find why I added it there)

However my opinion is that adding network to methods is not such a bad idea, network is useful to have around.

Reply:
As discussed this will stay as it is, and future redesign may take place.